Updating the search doc

dockerymick opened the pull request:
https://github.com/open-cluster-management/backlog/issues/7311

dockerymick committed:
Updating the search doc

dockerymick committed:
Update search.adoc

dockerymick committed:
added api file for search

dockerymick committed:
Merge branch '7311' of github.com:open-cluster-management/rhacm-docs into 7311

dockerymick commented on apis/search.json.adoc:
```diff
@@ -0,0 +1,403 @@
+=== Tags
+
+* search.open-cluster-management.io/v1beta1 : Create and manage persistence volumes
+
```
From review with Sherin Update v1beta1 to `v1`

dockerymick committed:
Update search.adoc

dockerymick committed:
updated the api version

dockerymick committed:
making more updates for Search

dockerymick committed:
updated the API

dockerymick committed:
Update search.adoc

Changes made based on the suggestion from Sherin

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
 = Search in the console
 
-For {product-title}, Search provides visibility into your resources across all your clusters.
+For {product-title}, search provides visibility into your resources across all of your clusters. Search uses Redisgraph (`redisgraph-tls`), an in-memory database to store the Kubernetes resources and the resource relationships. Create a `searchCustomization` custom resource (CR) to define the storage settings for search persistence. 
```
We need `searchcustomization` lower case, not camel case.

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
+For {product-title}, search provides visibility into your resources across all of your clusters. Search uses Redisgraph (`redisgraph-tls`), an in-memory database to store the Kubernetes resources and the resource relationships. Create a `searchCustomization` custom resource (CR) to define the storage settings for search persistence. 
 
-You can type any text value in the _Search box_ and results include anything with that value from any property, such as a name or namespace. If you search for `RedHat` and you manage a large number of clusters, it might take a longer time to receive search results.
+*Note:* If the `redisgraph-tls` pod crashes, data is resent from the `search-collector` pod to rebuild the cache.
```
Note not relevant in this context

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
+[#search-customization]
+== Search customization
+
+Update the storage settings for search by updating the `searchCustomization` CR. Search customization is namespace-scoped and located where search is installed in the hub cluster. By default, `searchCustomization` CRD is created.
```
We need searchcustomization lower case, not camel case.
Update the storage settings for search by creating the `searchcustomization` CR. 

> By default, `searchCustomization` CRD is created.

This need not be mentioned.

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
-For more specific search results, include the property selector in your search. You can combine related values for the property, for a more precise scope of your search. For example, search for `cluster:dev red` to receive results that match the string "red" in the `dev` cluster.
+[#search-customization]
+== Search customization
+
```
The default installation of ACM will be configured to persist search index. We can see the volume being used in the search-redisgraph statefulset definition with the volume name `persist`.

dockerymick replied:
@SherinV Here is my incoming update: When you install {product-title-short}, the product is configured to persist search index.

Should the article "the" come before search index? 

`...is configured to persist the search index? `

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
+
+Update the storage settings for search by updating the `searchCustomization` CR. Search customization is namespace-scoped and located where search is installed in the hub cluster. By default, `searchCustomization` CRD is created.
+
+Your cluster must have a default storage class defined. If your cluster has a defined default storage class, the search component creates a Persistent Volume Claim (PVC) of 10G on the storage class. Verify if there is a default storage class in your cluster by running the following command: `kubectl get storageclass`. 
```
> Your cluster must have a default storage class defined.

This is not a requirement as we fallback to emptyDir by default

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
+
+Your cluster must have a default storage class defined. If your cluster has a defined default storage class, the search component creates a Persistent Volume Claim (PVC) of 10G on the storage class. Verify if there is a default storage class in your cluster by running the following command: `kubectl get storageclass`. 
+
+If a default storage class is absent or search fails to create a PVC, search uses the `emptyDir` as a volume. You can disable use of the `emptyDir` by updating the `persistence` flag to `False` in the customization CR. Review the following scenarios when you update the `storageClass` flag and there is a failure to create the PVC:
```
You can disable persistence by updating the `persistence` flag to `false` in the customization CR which will turn off saving search index.

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
+
+* When the persistence flag is set to `True`:
+** If the `storageClass` is set, search uses the provided storage class.
+** If the `storageClass` is not set, search attempts to use the default storage class. If there is no default storage class, search uses the `emptyDir` volume.
```
> If there is no default storage class, search uses the emptyDir volume.

This is not true any more when persistence flag is set to `True`

SherinV commented on console/search.adoc:
```diff
@@ -1,11 +1,51 @@
+** The `storageClass` or `storageSize` has no effect.
+** Search disables persistence.
+
+A status for persistence can be retrieved from the search operator (`searchoperator`). For more information, see link:https://kubernetes.io/docs/concepts/storage/volumes/#emptydir[Volumes] in the Kubernetes documentation. View the following table to view status messages for the `searchCustomization` CR:
```
A status for persistence can be retrieved from the search operator (searchoperator) should suffice. The status should be self explanatory. We can remove the table.

dockerymick committed:
Update search.adoc

*Note:* If the `redisgraph-tls` pod crashes, data is resent from the `search-collector` pod to rebuild the cache.

By default, `searchcustomization` CRD is created.

Your cluster must have a default storage class defined.


For more information, see link:https://kubernetes.io/docs/concepts/storage/volumes/#emptydir[Volumes] in the Kubernetes documentation. View the following table to view status messages for the `searchcustomization` CR:

.Status table
|===
| Parameter | Status message
| statusUsingPVC | "Redisgraph is using PersistenceVolumeClaim"
| statusDegradedEmptyDir | "Degraded mode using EmptyDir. Unable to use PersistenceVolumeClaim"
| statusUsingNodeEmptyDir | "Node level persistence using EmptyDir"
| statusFailedDegraded | "Unable to create Redisgraph Deployment in Degraded Mode"
| statusFailedUsingPVC | "Unable to create Redisgraph Deployment using PVC"
| statusFailedNoPersistence | "Unable to create Redisgraph Deployment"
|===

dockerymick committed:
Merge branch '2.2_stage' into 7311

dockerymick committed:
Update search.adoc

 Changes during review with Sherin and Xavier 

Verify if there is a default storage class in your cluster by running the following command: `kubectl get storageclass`. 

If a default storage class is absent or search fails to create a PVC, search uses the `emptyDir` as a volume. 

See link:../apis/search.json.adoc#search-api[Search customization API] for more information.

dockerymick committed:
Update search.adoc

Final review of search file
